### PR TITLE
Implement str.translate()

### PR DESCRIPTION
**Pull request by 1e-to**

### CODE REVIEW

**Comment by 1e-to** [comment]
I have some questions.
1. Is it possible to somehow translate a dictionary into a method implementation without explicitly setting a numba dictionary?
2. Numba does not know how to work with dictionaries of strings of different types, an example is an "test_translate_error". Is there something I need to do in this case, or just ignore it? Also, Numba cannot create dictionaries with None, what should I do in this case?

**Comment by sklam** [comment]
1) Can you explain what you meant by "translate a dictionary into a method implementation"?

2) These are limitations of the current typed-dict implementation. Can you open an issue for your usecase? For now, you may want to skip such cases.

**Comment by 1e-to** [comment]
> 1. Can you explain what you meant by "translate a dictionary into a method implementation"?

In the function that I am implementing we get the parameter - dictionary. But I can not use the usual dictionary, a typing error occurs. I use numba.typed.Dict (), but may be another way?

**Comment by sklam** [comment]
`numba.typed.Dict()` is the current way. 

Inside jit region, the syntax `{'a': 1}` will make a `numba.typed.Dict`. But outside (in interpreted code), it would be a normal python dictionary.

**Comment by 1e-to** [comment]
Thank you for your responses, 
Now can you add this PR to queue on review?

File: numba/tests/test_unicode.py [hunk 1]
```diff
@@ -1171,6 +1176,54 @@ def test_istitle(self):
...
+        msg = 'Results "{}".translate({}) must be equal'
+        for s in UNICODE_EXAMPLES + cpython + extra:
+            for t in table + table_extra:
+                numba_t = numba.typed.Dict()
+                for key, value in t.items():
+                    numba_t[key] = value
+                self.assertEqual(pyfunc(s, t), cfunc(s, numba_t),
+                                 msg=msg.format(s, t))
+
+    @unittest.skip('Dont work in numba')
```
Comment (stuartarchibald) [inline numba/tests/test_unicode.py]: This needs resolving, what's the issue?

Comment (1e-to) [reply numba/tests/test_unicode.py]: https://github.com/numba/numba/issues/4873

**Comment by stuartarchibald** [comment]
Please could you resolve the conflicts by merging master into this branch. Thanks!

**Pull request closed without merging**